Changed the color for the table headings for enough contrast with the background

kivuvarosekivuvan opened the pull request:
# Engineers Checklist
## Description
- This PR addresses the contrast issue of the table heading with the background, making it black/the color specified in the design system.
- For the inactive state, we are using `--color-light-contrast` (typically Black).
- For the hover or active state, we are using `--color-primary-50.`


## Screenshot
<img width="1534" alt="Screenshot 2024-07-30 at 09 15 20" src="https://github.com/user-attachments/assets/a52ec2ef-af0f-41e6-bc9b-ab65eb25ad28">


## Testing
- The changes have been tested using the WAVE tool and a color contrast checker to verify the improved contrast.

Commit:
Changed the color for the table headings for enough contrast with the background

kivuvarosekivuvan commented:
@Angamanga I have changed to black but waiting for Reuben's thoughts on hover state for the `<th>`

Ifycode commented on apps/web-mzima-client/src/app/settings/users/users.component.scss:
```diff
@@ -38,6 +38,18 @@
+    //     }
+    // .p-datatable .p-sortable-column.p-highlight .p-sortable-column-icon{
+    //   color: var(--color-light-contrast);
+    // }
```
You can remove the comment to clean this up @kivuvarosekivuvan 
Or is there a reason this this here?

kivuvarosekivuvan replied:
This was just a test, I will get rid of that

Ifycode commented on apps/web-mzima-client/src/app/settings/users/users.component.scss:
```diff
@@ -38,6 +38,18 @@
+    }
+    .p-datatable .p-sortable-column .p-sortable-column-icon {
+      color: var(--color-light-contrast);
+    }
```
Since this is an `scss` setting, let's use `scss` method instead of the regular `css` method of accessing children elements @kivuvarosekivuvan

kivuvarosekivuvan replied:
@Ifycode You mean accessing children using the ampersand operator(&)?

kivuvarosekivuvan commented:
For this PR, I’m waiting for reuben’s approval of the colors. I will use
scss when he approves the colors

On Mon, 29 Jul 2024 at 16:05 Mary @Ifycode ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In apps/web-mzima-client/src/app/settings/users/users.component.scss
> <https://github.com/ushahidi/platform-client-mzima/pull/1160#discussion_r1695175243>
> :
>
> > @@ -38,6 +38,18 @@
>          }
>        }
>      }
> +    //     .p-datatable .p-sortable-column.p-highlight{
> +    // color: var(--color-light-contrast);
> +    //     }
> +    // .p-datatable .p-sortable-column.p-highlight .p-sortable-column-icon{
> +    //   color: var(--color-light-contrast);
> +    // }
>
> You can remove the comment to clean this up @kivuvarosekivuvan
> <https://github.com/kivuvarosekivuvan>
> Or is there a reason this this here?
> ------------------------------
>
> In apps/web-mzima-client/src/app/settings/users/users.component.scss
> <https://github.com/ushahidi/platform-client-mzima/pull/1160#discussion_r1695186515>
> :
>
> > @@ -38,6 +38,18 @@
>          }
>        }
>      }
> +    //     .p-datatable .p-sortable-column.p-highlight{
> +    // color: var(--color-light-contrast);
> +    //     }
> +    // .p-datatable .p-sortable-column.p-highlight .p-sortable-column-icon{
> +    //   color: var(--color-light-contrast);
> +    // }
> +    .p-datatable .p-datatable-thead > tr > th {
> +      color: var(--color-light-contrast);
> +    }
> +    .p-datatable .p-sortable-column .p-sortable-column-icon {
> +      color: var(--color-light-contrast);
> +    }
>
> Since this is an scss setting, let's use scss method instead of the
> regular css method of accessing children elements @kivuvarosekivuvan
> <https://github.com/kivuvarosekivuvan>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ushahidi/platform-client-mzima/pull/1160#pullrequestreview-2204874684>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5TCBSLL6BP7FYWS5GBN3UTZOY4X3AVCNFSM6AAAAABKYWPLASVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDEMBUHA3TINRYGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Commit:
Improve table header contrast and styling

- Set table header text color to --color-light-contrast for inactive state
- Use --color-primary-50 for hover or active state on sortable columns
- Addresses contrast issue between table header and background
- Used SCSS to access child components and improve code readability

kivuvarosekivuvan commented:
@Ifycode This is ready for review. Changes in already :tada:

Commit:
Used scss to style the new colors of the table headings

Commit:
Remove .vscode/settings.json file
